task_sdk/src/airflow/sdk/definitions/baseoperator.py

```diff
@@ -0,0 +1,1141 @@
+    :param logger_name: Name of the logger used by the Operator to emit logs.
+        If set to `None` (default), the logger name will fall back to
+        `airflow.task.operators.{class.__module__}.{class.__name__}` (e.g. SimpleHttpOperator will have
+        *airflow.task.operators.airflow.providers.http.operators.http.SimpleHttpOperator* as logger).
+    :param allow_nested_operators: if True, when an operator is executed within another one a warning message
+        will be logged. If False, then an exception will be raised if the operator is badly used (e.g. nested
+        within another one). In future releases of Airflow this parameter will be removed and an exception
+        will always be thrown when operators are nested within each other (default is True).
+
+        **Example**: example of a bad operator mixin usage::
+
+            @task(provide_context=True)
+            def say_hello_world(**context):
+                hello_world_task = BashOperator(
+                    task_id="hello_world_task",
+                    bash_command="python -c \"print('Hello, world!')\"",
+                    dag=dag,
+                )
+                hello_world_task.execute(context)
+    """
+
+    task_id: str
+    owner: str = DEFAULT_OWNER
+    email: str | Sequence[str] | None = None
+    email_on_retry: bool = True
+    email_on_failure: bool = True
+    retries: int | None = DEFAULT_RETRIES
+    retry_delay: timedelta | float = DEFAULT_RETRY_DELAY
+    retry_exponential_backoff: bool = False
+    max_retry_delay: timedelta | float | None = None
+    start_date: datetime | None = None
+    end_date: datetime | None = None
+    depends_on_past: bool = False
+    ignore_first_depends_on_past: bool = DEFAULT_IGNORE_FIRST_DEPENDS_ON_PAST
+    wait_for_past_depends_before_skipping: bool = DEFAULT_WAIT_FOR_PAST_DEPENDS_BEFORE_SKIPPING
+    wait_for_downstream: bool = False
+    dag: DAG | None = None
+    params: collections.abc.MutableMapping | None = None
+    default_args: dict | None = None
+    priority_weight: int = DEFAULT_PRIORITY_WEIGHT
+    # TODO:
+    weight_rule: PriorityWeightStrategy | str = DEFAULT_WEIGHT_RULE
+    queue: str = DEFAULT_QUEUE
+    pool: str = "default"
+    pool_slots: int = DEFAULT_POOL_SLOTS
+    execution_timeout: timedelta | None = DEFAULT_TASK_EXECUTION_TIMEOUT
+    # on_execute_callback: None | TaskStateChangeCallback | list[TaskStateChangeCallback] = None
+    # on_failure_callback: None | TaskStateChangeCallback | list[TaskStateChangeCallback] = None
+    # on_success_callback: None | TaskStateChangeCallback | list[TaskStateChangeCallback] = None
+    # on_retry_callback: None | TaskStateChangeCallback | list[TaskStateChangeCallback] = None
+    # on_skipped_callback: None | TaskStateChangeCallback | list[TaskStateChangeCallback] = None
+    # pre_execute: TaskPreExecuteHook | None = None
+    # post_execute: TaskPostExecuteHook | None = None
```

Comment:
This are still experimental -- wonder if we should just kill it.

Reply:
Yes perhaps. The reason I've left it is that this PR is **trying** to just be about definition time, and while these are set at defn time this is much more about execution time.

Reply:
For the future (understand runtime is mostly out of scope for this pr): Are these used/to be used by OpenLineage, or would it keep listening for task status change?

Reply:
Nope, openlineage uses "Listeners" which are global to the deployment, not specific to one dag or task.